lib/timescaledb/schema_dumper.rb

```diff
@@ -61,17 +61,33 @@ def timescale_retention_policy(hypertable, stream)
 
     def timescale_compression_settings_for(hypertable)
       compression_settings = hypertable.compression_settings.each_with_object({}) do |setting, compression_settings|
-        compression_settings[:compress_segmentby] = setting.attname if setting.segmentby_column_index
+        # It's possible to configure compression so that it is segmented by multiple
+        # columns. To make sure we capture that correctly, we'll treat them as an array.
+        compression_settings[:compress_segmentby] ||= []
+        compression_settings[:compress_orderby] ||= []
+
+        compression_settings[:compress_segmentby] << setting.attname if setting.segmentby_column_index
 
         if setting.orderby_column_index
           direction = setting.orderby_asc ? "ASC" : "DESC"
-          compression_settings[:compress_orderby] = "#{setting.attname} #{direction}"
+          direction += " NULLS FIRST" if setting.orderby_asc && setting.orderby_nullsfirst
+          direction += " NULLS LAST" if !setting.orderby_asc && !setting.orderby_nullsfirst
```

Comment:
Good catch!

I'm not sure about why the `orderby_asc` should be considered again here. Can we simplify it to use the `orderby_nullfirst` only?

```suggestion
          null_direction = setting.orderby_nullsfirst ? "FIRST" : "LAST"
          direction += " NULLS #{null_direction}"
```

Reply:
According to the PG docs, NULLS LAST is default for ASC ordering, and NULLS FIRST is default for DESC ordering.

If we want to be explicit, we can do it your way - but I was trying to keep the default approach.